src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2846,6 +2846,14 @@ private bool TryGetReturnType(out TypeWithAnnotations type, out FlowAnalysisAnno
         public override BoundNode? VisitLocal(BoundLocal node)
         {
             var local = node.LocalSymbol;
+
+            // Avoid cycles in nullable analysis.
+            if (local is SourceLocalSymbol { IsVar: true } && local.ForbiddenZone?.Contains(node.Syntax) == true)
```

Comment:
Can we document how exactly this helps avoid cycles?

Reply:
Good idea, added: ae067f41afe43f44aa062bf8c31178b112e060ed